### ISSUE TITLE
Prototype Pollution Vulnerability - Veracode

### ISSUE THREAD

**Issue by fa-bcal** [ag-grid/ag-grid#8271]
**I'm submitting a ...** 
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```
[NVD](https://nvd.nist.gov/vuln/detail/CVE-2024-39001) record

Veracode scan has produced 2 High Vulnerability flags.
1. [CVE-2024-39001](http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2024-39001)
2. [CVE-2024-38996](http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2024-38996)

Prototype Pollution: ag-grid-enterprise is vulnerable to Prototype Pollution. The vulnerability is due to the functions `_.mergeDeep`, `_ModuleSupport.jsonApply`, `_ModuleSupport.setPath`, and `_Util.jsonApply` accepting arguments that include the built-in property `proto`. Attackers can exploit this by passing specially crafted arguments to these functions.

![image](https://github.com/ag-grid/ag-grid/assets/174833997/1e61e17d-d4bd-47d2-9787-22195be7656b)

* **AG Grid version:** 32.0
This issue has been flagged as present in all versions including 32.0

**Comment by AG-Zoheil** [comment]
Hi @fa-bcal ,

Thank you for raising this on GitHub. Please note that we are aware of the issue and are looking to release a patch for version 32 this week which will include a fix for this.

Kind regards,
Zoheil

**Comment by jpsla94** [comment]
Hello @AG-Zoheil,

Could you kindly update this topic once the fix is applied?

Best regards,
João Lola

**Comment by AG-Zoheil** [comment]
Hi @jpsla94,

We are looking to release the patch by tomorrow.

However the best way to track this is to sign up for AG Grid new release notifications using [the instructions here](https://ag-grid.zendesk.com/hc/en-us/articles/360002648058-How-do-I-sign-up-for-ag-Grid-release-notifications-). This way you'll know as soon as a new version is out and you can check whether this specific item was implemented on [the changelog page](https://www.ag-grid.com/ag-grid-changelog/).

Kind regards,
Zoheil

**Comment by jpsla94** [comment]
@AG-Zoheil Thank you.

**Comment by noren95** [comment]
Hi @AG-Zoheil 
Can you confirm that all versions prior the expected patch release are vulnerable?
This advisory marks only one: https://github.com/advisories/GHSA-876p-c77m-x2hc
Thank you!

**Comment by AG-Zoheil** [comment]
AG Grid has resolved this security vulnerability in `v32.0.1` which is now available. Please upgrade to this latest version.

**Comment by noren95** [comment]
@AG-Zoheil Thank you for the update, 
In addition to my previous question above I would like you to confirm:
1. The fix applied to ag-grid-enterprise and ag-grid-community and solves CVE-2024-38996.
2. The fix applies to CVE-2024-39001, only for ag-grid-enterprise. if not, is there a fix for this one?
